src/llmq/quorums_signing.cpp

```diff
@@ -36,14 +36,46 @@ bool CRecoveredSigsDb::HasRecoveredSig(Consensus::LLMQType llmqType, const uint2
 
 bool CRecoveredSigsDb::HasRecoveredSigForId(Consensus::LLMQType llmqType, const uint256& id)
 {
+    int64_t t = GetTimeMillis();
+
+    auto cacheKey = std::make_pair(llmqType, id);
+    {
+        LOCK(cs);
+        auto it = hasSigForIdCache.find(cacheKey);
+        if (it != hasSigForIdCache.end()) {
+            it->second.second = t;
+            return it->second.first;
```

Comment:
why do we store `bool` i.e. why can't we simply return `true` here?

same for `Session` variant

Reply:
Because the cache also holds the results for `HasXXX` calls that returned `false`. If we wouldn't store the `false` case in the cache, the whole cache optimization would be ineffective due to the many calls where `false` is the most-returned value.